assets/src/scripts/components/graph/state/well-log.js

```diff
@@ -192,10 +192,23 @@ export const getConstructionElements = memoize((opts, chartType) => createSelect
     (elements, xScale, yScale, selectedId) => {
         const parts = elements.map(element => {
             const loc = element.position.coordinates;
+            // Visualize well construction components that have no listed length
+            // If the element has no listed length (for example depth 77.0-77.0) add 6 inches (0.5 ft).
+            // This will not show in the well table but will show as an element in the well diagram.
+            // First check if the element has marked as zeroLength, if so skip this step
+           if (!element.zeroLength && element.position.coordinates.start === element.position.coordinates.end ) {
+                element.position.coordinates.end = element.position.coordinates.end  + 0.5;
+                // Add an attribute to indicate that this element is of zero length and has been adjusted.
+                // This is needed to prevent the 'locString' from being over written when this function is called again (and it will be called)
+                element.zeroLength = true;
```

Comment:
Does this change what is stored in the state? If not, I don't understand this comment. If it does, that violates how we should use redux ... state is only changed through actions, selectors should only get state information.

Reply:
Ah, yes. You have a point. I'll have to ponder this for a bit.

Reply:
Okay, I think I have the fixed.